lahja/asyncio/endpoint.py

```diff
@@ -240,12 +200,17 @@ def __init__(self, name: str) -> None:
             )
 
         # Signal when a new remote connection is established
-        self._remote_connections_changed = asyncio.Condition()
+        self._remote_connections_changed = asyncio.Condition()  # type: ignore
+
+        # Signal when at least one remote has had a subscription change.
+        self._remote_subscriptions_changed = asyncio.Condition()  # type: ignore
 
         # storage containers for inbound and outbound connections to other
         # endpoints
-        self._full_connections = {}
-        self._half_connections = set()
+        self._connections = set()
```

Comment:
*nit*: `is_connected_to` and `_get_remote` would both be easier to write if this was a `Dict[remote.name, remote]`. I'm not sure whether that's worth making the *other* lines which use `_connections` more complicated but wanted to point it out.

Reply:
Yup, that change works now that remotes always have names at the time we put them in this data structure which wasn't previously the case.  Will change.

Reply:
I'm actually not going to do this.  We can tackle it later if it proves to be a performance issue, however, I have grown to like that we have a single canonical location for endpoint names and to store it as a dict would allow for the `remote.name` and the name it's stored under to diverge.